google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerSchemaManagementTool.java

```diff
@@ -52,4 +55,15 @@ SpannerTableExporter getSpannerTableExporter(ExecutionOptions options) {
     JdbcContext jdbcContext = this.resolveJdbcContext(options.getConfigurationValues());
     return (SpannerTableExporter) jdbcContext.getDialect().getTableExporter();
   }
+
+  SpannerDatabaseInfo getDatabaseMetaData(ExecutionOptions options) {
```

Comment:
How expensive is this operation? Does Hibernate already do something like this in the schema update mode? Can we re-use their logic?

Reply:
How expensive is this operation?
- It is not expensive, the added runtime is trivial. Hibernate does this on every update operation.

Does Hibernate already do something like this in the schema update mode?
- Yes, the code I got for doing this came out of Hibernate. However we have to retrieve the metadata info again, I was not able to find a way to retrieve the existing information.

Can we re-use their logic?
- Yup, this is copied over here, but I wasn't able to find a way to reuse the information they computed.